src/Apprentice.Bot.Connectors/Clients/DirectLineClient.cs

```diff
@@ -23,9 +24,32 @@ public DirectLineClient(Uri botAddress, string authToken)
                 new AuthenticationHeaderValue("Bearer", authToken);
         }
 
-        public Task<HttpResponseMessage> StartConversationAsync()
+        public async Task<HttpResponseMessage> StartConversationAsync()
         {
-            return this.client.GetAsync("/v3/directline/conversations");
+            var newToken = await GenerateToken();
+
+            this.client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", newToken);
+            return await this.client.GetAsync("/v3/directline/conversations");
+        }
+
+        private async Task<string> GenerateToken()
+        {
+            var content = new StringContent("");
+            content.Headers.ContentType.MediaType = "application/json";
+
+            var result = await this.client.PostAsync($"/v3/directline/tokens/generate", content);
+
+            result.EnsureSuccessStatusCode();
+
+            dynamic json = await result.Content.ReadAsStringAsync();
+            GenerateTokenResponse tokenResult = JsonConvert.DeserializeObject<GenerateTokenResponse>(json);
+
+            if (tokenResult.Token == null)
```

Comment:
Might be worthwhile using the null conditional operator here...
```suggestion
            if (tokenResult?.Token == null)
```

Reply:
OK